txpipe/twopoint.py

```diff
@@ -564,7 +565,113 @@ def read_metadata(self):
 
         return meta
 
-    
+# Aperture Mass class that inherits from TXTwoPoint
+class TXApertureMass(TXTwoPoint):
+
+    name='TXApertureMass'
+    inputs = [
+        ('calibrated_shear_catalog', ShearCatalog),
+        ('calibrated_lens_catalog', HDFFile),
+        ('shear_photoz_stack', HDFFile),
+        ('patch_centers', TextFile),
+        ('tracer_metadata', HDFFile),
+    ]
+    outputs = [
+        ('aperturemass_data', SACCFile),
+    ]
+    # Add values to the config file that are not previously defined
+    config_options = {
+        'calcs': [0,1,2],
+        'min_sep': 0.5,
+        'max_sep': 300.,
+        'nbins': 15, # MEAD: Changed default from 9 -> 15
+        'bin_slop': 0.02,
+        'sep_units': 'arcmin',
+        'flip_g1': False,
+        'flip_g2': True,
+        'cores_per_task': 20,
+        'verbose': 1,
+        'source_bins': [-1],
+        'lens_bins': [-1],
+        'reduce_randoms_size': 1.0,
+        'var_method': 'jackknife',
+        'use_true_shear': False,
+        'subtract_mean_shear': False,
+        'use_randoms': False,
+        'low_mem': False,
+        }
+
+    def select_calculations(self, source_list, lens_list):
+
+        # For shear-shear we omit pairs with j>i
+        calcs = []
+        k = SHEAR_SHEAR
+        for i in source_list:
+            for j in range(i+1):
+                if j in source_list:
+                    calcs.append((i,j,k))
+
+        if self.rank==0:
+            print(f"Running these calculations: {calcs}")
+
+        return calcs
+
+    def calculate_shear_shear(self, i, j):
+
+        gg = TXTwoPoint.calculate_shear_shear(self, i, j)
```

Comment:
Can use `super()` here instead of `TXTwoPoint` in case we e.g. change the name of the parent class or otherwise modify the structure later.

Reply:
Okay, I think I've implemented `super()` correctly now.